lib/_stream_duplex.js

```diff
@@ -66,6 +66,11 @@ function Duplex(options) {
   }
 }
 
+Duplex.prototype.acquireStandardReadableStream =
+  Readable.prototype.acquireStandardStream;
+Duplex.prototype.acquireStandardWritableStream =
+  Writable.prototype.acquireStandardStream;
```

Comment:
There's no concept of a duplex in WHATWG streams so I can't add support for it.

Reply:
I think you're missing the point.

(a) You're adding methods to the `Duplex` prototype that are not included in the documentation edits above

(b) This implementation creates an incongruency in the `Writable`, `Readable`, `Duplex`, `Transform` API that needs to be either reconciled or documented. Specifically:

```js
const { Writable, Readable, Duplex, Transform } = require('stream')
const w = new Writable({ write() {} })
const r = new Readable({ read() {} })
const d = new Duplex({ read() {}, write() {} })
const t = new Transform({ transform() {} });

w.acquireStandardStream() // Returns WritableStream
r.acquireStandardStream() // Returns ReadableStream
d.acquireStandardStream() // Returns ReadableStream
t.acquireStandardStream() // Returns TransformStream
```

Given that all `Duplex` streams are both `Writable` and `Readable`, this API sets things up so that the user cannot treat them consistently. That is, I cannot blindly treat a `Duplex` as a `Writable` when it comes to calling the `acquireStandardStream()` method. Also, while a Transform is a Duplex, it's implementation of `acquireStandardStream()` is different than all other Duplex instances.

That's just bad API design.

What would be easier, and significantly simpler, would be something like:

```js
const stream = require('stream')
const { toReadableStream, toWritableStream } = stream
const { Writable, Readable, Duplex, Transform } = stream
const w = new Writable({ write() {} })
const r = new Readable({ read() {} })
const d = new Duplex({ read() {}, write() {} })
const t = new Transform({ transform() {} });

toReadableStream(r) // ReadableStream
toReadableStream(d) // ReadableStream
toReadableStream(t) // ReadableStream

toWritableStream(w) // WritableStream
toWritableStream(d) // WritableStream
toWritableStream(t) // WritableStream
```

This would have the added benefit of not adding methods to the prototype so that it would Just Work (tm) for `readable-streams`.

Reply:
> You're adding methods to the Duplex prototype that are not included in the documentation edits above

whoops, i'll add those right away.

> `d.acquireStandardStream() // Returns ReadableStream`

Does deleting `Duplex.prototype.acquireStandardStream` not solve this problem?

Reply:
That does not really solve the issue. Users will end up having to check everything to see what kind of Readable or Writable it is, rather than just treating those all the same. That is, I would have to do `const w = thing instanceof stream.Duplex ? thing.acquireStandardWritableStream() : thing.acquireStandardStream()` if there's any chance at all that the thing they could be getting is a `Duplex`... which it quite often will be (`process.stdout, process.stdin, net.Socket, http2.Http2Stream`, and so forth)... It's far better to be able to use a single consistent method for all Writables and all Readables.